lib/puppet/defaults.rb

```diff
@@ -419,7 +419,20 @@ def self.default_diffargs
     :data_binding_terminus => {
       :type    => :terminus,
       :default => "hiera",
-      :desc    => "Where to retrieve information about data.",
+      :desc    =>
+        "This setting has been deprecated. Use of any value other than 'hiera' should instead be configured
+         in a version 5 hiera.yaml. Until this setting is removed, it controls which data binding terminus
```

Comment:
@hlindberg do you know where the differences between hiera.yaml versions are documented? I looked on our docs site but did not find them.